tools/cpp/windows_cc_configure.bzl

```diff
@@ -254,17 +249,19 @@ def find_vc_path(repository_ctx):
     auto_configure_warning_maybe(repository_ctx, "Visual C++ build tools found at %s" % vc_dir)
     return vc_dir
 
-def _is_vs_2017_or_2019(repository_ctx, vc_path):
-    """Check if the installed VS version is Visual Studio 2017 or 2019."""
+def _is_vs_2017_or_newer(repository_ctx, vc_path):
+    """Check if the installed VS version is Visual Studio 2017 or newer."""
 
-    # The layout of VC folder in VS 2017 and 2019 is different from that in VS 2015 and older versions.
-    # In VS 2017 and 2019, it contains only three directories:
+    # The layout of VC folder in VS 2017 and newer versions is different from that in VS 2015 and older versions.
+    # From VS 2017 it contains three directories:
     # "Auxiliary", "Redist", "Tools"
+    # From VS 2022 17.6, a fourth "vcpkg" directory is also present, but we only check presence of the three
+    # directories above in case other directories pop up in the future
 
-    vc_2017_or_2019_contents = ["auxiliary", "redist", "tools"]
+    vc_2017_or_newer_contents = ["auxiliary", "redist", "tools"]
     vc_path_contents = [d.basename.lower() for d in repository_ctx.path(vc_path).readdir()]
-    vc_path_contents = sorted(vc_path_contents)
-    return vc_path_contents == vc_2017_or_2019_contents
+    vc_path_contents = sorted([d for d in vc_path_contents if d in vc_2017_or_newer_contents])
```

Comment:
I guess we can simplify the logic here:

```
vc_2017_or_newer_expected_dirs = ["auxiliary", "redist", "tools"]
for dir in vc_2017_or_newer_expected_dirs:
    if not repository_ctx.path(vc_path).get_child(dir).exists:
        return False
return True
```

Reply:
@redsun82 Can you overtake the suggested change - would be nice if we could soon merge this in bring it to Bazel 6.3.0

Reply:
@redsun82 Can you overtake the suggested change - would be nice if we could soon merge this in bring it to Bazel 6.3.0

Reply:
I can fork this and add @meteorcloudy  suggestion if @redsun82 does not have time/resources to work on this further...

Reply:
I create a PR with this change -> https://github.com/bazelbuild/bazel/pull/18945 @meteorcloudy, @redsun82